s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/S3SourceTask.java

```diff
@@ -133,11 +137,12 @@ private void initializeS3Client() {
 
     private void prepareReaderFromOffsetStorageReader() {
         sourceRecordIterator = new SourceRecordIterator(s3SourceConfig, s3Client, this.s3Bucket, offsetManager,
-                this.outputWriter, failedObjectKeys);
+                this.transformer, fileReader);
     }
 
     @Override
     public List<SourceRecord> poll() throws InterruptedException {
+        LOGGER.info("Polling again");
         synchronized (pollLock) {
```

Comment:
So in case of big files we will be blocked until we read the whole file?

Reply:
Kind of yes. We have this kind of pollLock in our jdbc source connector too.

Reply:
Will be handled in ticket https://aiven.atlassian.net/browse/KCON-28